pallets/pallet-deposit-storage/src/lib.rs

```diff
@@ -171,6 +174,20 @@ pub mod pallet {
 		DepositEntryOf<T>,
 	>;
 
+	/// Storage of all system deposits. They are the same as user deposits, but
+	/// cannot be claimed back by the payers. Instead, some on chain logic must
+	/// trigger their release.
+	#[pallet::storage]
+	#[pallet::getter(fn system_deposits)]
+	pub(crate) type SystemDeposits<T> = StorageDoubleMap<
```

Comment:
Changing the existing storage would require a migration. I think it's anyway better to divide between "system" deposits which can only be released by the runtime and "regular" deposits that can be claimed back by the user.

Reply:
Fine by me! I don't know much about the architecture of the deposit pallet, but if hold reasons are sufficiently informative to identify the pallet, and if there's a hook that's executed whenever someone tries to reclaim a deposit through the pallet (instead of the release being triggered by another pallet), then I could also see that this may be configured to simply fail when the deposit relates to the bonding curve pallet (for example) _OR_ to try and call `finish_destroy` there, which would lead to the deposit being reclaimed unless the pool is not ready to be destroyed.

Reply:
Sorry I am not sure I am able to fully understand your message. If pallets interact with the deposit storage pallet via the `MutateHold` trait, then any deposit that needs to be removed would have to go through some other runtime components. The deposit storage pallet ALSO exposes one extrinsic of its own, which allows a deposit to be reclaimed back by the user. To be honest, I would simply, in the future, remove such an extrinsic and "force" all pallets to interact with the deposit storage via simply the `MutateHold` trait, without any complex hooks and similar. So to your point, if you call `T::MutateHold::release(<right_unique_reason>, <user_address>)`, it should be enough.